src/security/cert_generators/file/security_file_certgen.cc

```diff
@@ -276,29 +292,38 @@ int main(int argc, char *argv[])
             }
         }
 
+        // when -s is used, -M is required
+        if (!db_path.empty() && max_db_size == 0)
+            throw std::runtime_error("security_file_certgen is missing the required parameter. There should be -s and -M parameters together, or neither used.");
+
         if (create_new_db) {
+            // when -c is used, -s is required (implying also -M, which is checked above)
+            if (db_path.empty())
+                throw std::runtime_error("security_file_certgen is missing the required parameter. There should be -s and -M parameters when -c is used.");
```

Comment:
Please consider this simpler version:

```
"security_file_certgen -c requires -s and -M parameters"
```

FWIW, I am not sure we should interpret `-c` as "initialize on-disk db". I suspect it is best interpreted as "initialize whatever is needed to be initialized". The latter interpretation would require renaming `create_new_db` to `initialize` and adding `!db_path.empty()` condition to this `if` statement. I think it is fine for a lonely `-c` to do nothing (for now).

Reply:
nod, I have the same idea for initialization in mind. But come to the opposite conclusion...

Allowing a lonely -c parameter to be used with the current helper version would make these helpers which do not support that behaviour return a "success" output if they are incorrectly attempted to be used to initialize a future type of DB cache not using -s. Leaving the admin with the wrong impression that it is initialized and puzzling over the subsequent behaviour and/or errors.

So I prefer to validate the documented parameters strictly according to the documented syntax in the man(8) page SYNOPSIS. That way we can rely on installations using those parameters only in ways the documentation defines as valid when changing them for future feature additions.